R/final_size.R

```diff
@@ -1,54 +1,77 @@
+#' @param contact_matrix  Social contact matrix. Entry $mm_{ij}$ gives average
+#' number of contacts in group $i$ reported by participants in group j
+#' @param demography_vector  Demography vector. Entry $pp_{i}$ gives proportion
+#' of total population in group $i$ (model will normalise if needed)
+#' @param prop_suscep  Proportion of each group susceptible. Null assumption is
+#' fully susceptible
 #' @keywords epidemic model
 #' @export
 #' @examples
-#' final_size()
+#' library(socialmixr)
+#' data(polymod)
+#' contact_data <- contact_matrix(polymod, countries = "United Kingdom", age.limits = c(0,20,40))
+#' 
+#' c_matrix <-t(contact_data$matrix) # Define contact matrix (entry ij is contacts in group i reported by group j)
+#' d_vector <- contact_data$participants$proportion # Define proportion in each age group
+#' p_suscep <- c(1,0.5,0.5) # Define proportion of age group that is susceptible to infection
+#' 
+#' # Run final size model
+#' final_size(r0=2,contact_matrix = c_matrix, demography_vector = d_vector, prop_suscep = p_suscep)
+#'
+final_size <- function(r0 = 2, contact_matrix, demography_vector,
+                       prop_suscep = NULL) {
 
-final_size<-function(r0=2,contact_matrix,demography_vector,prop_suscep=NULL){
-  
   # Check inputs
-  if(is.null(prop_suscep)){prop_suscep <- demography_vector*0 + 1} # Assume fully susceptible if no entry
-  if(length(contact_matrix[1,])!=length(demography_vector)){stop("demography vector needs to be same size as contact matrix")}
-  if(length(demography_vector)!=length(prop_suscep)){stop("demography vector needs to be same size as susceptibility vector")}
-  pp0=as.numeric(demography_vector/sum(demography_vector))
-  
+  if (is.null(prop_suscep)) {
+    prop_suscep <- demography_vector * 0 + 1
+  } # Assume fully susceptible if no entry
+  if (length(contact_matrix[1, ]) != length(demography_vector)) {
+    stop("demography vector needs to be same size as contact matrix")
+  }
+  if (length(demography_vector) != length(prop_suscep)) {
+    stop("demography vector needs to be same size as susceptibility vector")
+  }
+  pp0 <- as.numeric(demography_vector / sum(demography_vector))
+
   # Scale next generation matrix so max eigenvalue=r0
   mm0 <- as.matrix(contact_matrix)
-  mm0 <- r0*mm0/max(Re(eigen(mm0)$values))
-  
+  mm0 <- r0 * mm0 / max(Re(eigen(mm0)$values))
+
   # Define transmission matrix A = mm_ij*pp_j/pp_i
-  beta1 <- (mm0* prop_suscep) /pp0
-  beta2 <- t(t(beta1)*pp0)
-  
+  beta1 <- (mm0 * prop_suscep) / pp0
+  beta2 <- t(t(beta1) * pp0)
+
   # Newton method for solving final size equation: A(1-x) = -log(x)
-  
+
   # Define functions f and f'
   vsize <- length(pp0)
-  f1 <- function(beta2,x){ beta2%*%(1-x) +log(x) }
-  f2 <- function(beta2,x,size){-beta2+diag(size)/x }
+  f1 <- function(beta2, x) {
+    beta2 %*% (1 - x) + log(x)
+  }
+  f2 <- function(beta2, x, size) {
+    -beta2 + diag(size) / x
+  }
```

Comment:
Since calculations are a little bit involved, it would be very helpful to have some details on the calculations themselves in @details for user-facing info, and maybe also as comments to the code if there are non-trivial strategies employed, which aren't useful for the user to know, but which future code contributors might want to know about?

Reply:
That's a good point - again I'm not sure whether the function will look like this even in the short term. I also don't really understand the strategy used here in terms of solving the final size equation - perhaps somebody wants to add this with the correct level of detail?

Reply:
Addressed in https://github.com/epiverse-trace/finalsize/pull/19/commits/de61ad0268080eb149fd51dd6ce8369089abf75d